internal/dataplane/translator/translator.go

```diff
@@ -180,6 +180,13 @@ func (t *Translator) BuildKongConfig() KongConfigBuildingResult {
 		t.registerSuccessfullyTranslatedObject(&result.Consumers[i].K8sKongConsumer)
 	}
 
+	// genereate vaults
+	// REVIEW: add check for Kong gateway enterprise here?
```

Comment:
```suggestion
	// generate vaults
	// REVIEW: add check for Kong gateway enterprise here?
```
You can try and push Enterprise-only entities to a non-Enterprise instance all you want and all you'll do is lock up your config. OSS or free mode won't accept them. IIRC we're not doing Enterprise checks for anything else similar, are we?

Trying to do would probably introduce some unexpected complexity since there's not really any guarantee that the license state is stable or consistent, and I don't think we ever check the root endpoint after startup.

Reply:
done.